clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -71,6 +71,13 @@ public FileRecords(File file,
         this.isSlice = isSlice;
         this.size = new AtomicInteger();
 
+        // set the initial size of the buffer
+        resize();
```

Comment:
I think we might not have wanted this change. Since this is the only usage, inlining `resize` seems a little better.